skiko/src/jvmTest/kotlin/org/jetbrains/skiko/RendezvousBroadcastChannelTest.kt

```diff
@@ -71,6 +66,30 @@ class RendezvousBroadcastChannelTest {
         assertEquals(listOf(1, 1, 1, 1, 1), actualValues)
     }
 
+    @Test(timeout = 30000)
+    fun `multithreading sending and receiving should not cause deadlock`() {
+        val channel = RendezvousBroadcastChannel<Int>()
+
+        runBlocking {
+            val receiverJobs = (1..10).map {
+                launch(Dispatchers.IO) {
+                    repeat(1000) {
+                        channel.receive()
+                    }
+                }
+            }
+
+            val sendingJob = launch(Dispatchers.IO) {
```

Comment:
There shall be several concurrent senders in test as well.

Reply:
The class is internal, and we don't use it in such case